BHoM_UI/Components/Adapter/Move.cs

```diff
@@ -67,13 +68,30 @@ public class MoveCaller : MethodCaller
         [Input("pushConfig", "Config options for the Push to Target adapter")]
         [Input("active", "Execute the Move")]
         [Output("success", "Define if the Move was successful")]
-        public static bool Move(BHoMAdapter source, BHoMAdapter target, IRequest request = null, Dictionary<string, object> pullConfig = null, Dictionary<string, object> pushConfig = null, bool active = false)
+        public static bool Move(BHoMAdapter source, BHoMAdapter target, IRequest request = null, 
+            PullType pullType = PullType.AdapterDefault, ActionConfig pullConfig = null,
+            PushType pushType = PushType.AdapterDefault, ActionConfig pushConfig = null, bool active = false)
         {
+            if (source == null || target == null)
+            {
+                Engine.Reflection.Compute.RecordError("Adapter input cannot be null.");
+                return false;
+            }
+
+            // ---------------------------------------------//
+            // Mandatory Adapter Action set-up              //
+            //----------------------------------------------//
+            // The following are mandatory set-ups to be ALWAYS performed 
+            // before the Adapter Action is called,
+            // whether the Action is overrided at the Toolkit level or not.
+
```

Comment:
This is the reason why we want to have some initialisation in the UI itself rather than later in the adapter.
We allow for Action overloading and we want to assure that certain initialisation are done no matter the overload.